src/latexify/codegen/function_codegen.py

```diff
@@ -536,6 +536,63 @@ def visit_IfExp(self, node: ast.IfExp) -> str:
         latex += self.visit(node)
         return latex + r", & \mathrm{otherwise} \end{array} \right."
 
+    def visit_Match(self, node: ast.Match) -> str:
+        """Visit a match node"""
+        latex = r"\left\{ \begin{array}{ll} "
+        subject_latex = self.visit(node.subject)
+        for i, match_case in enumerate(node.cases):
+            if len(match_case.body) != 1:
+                raise exceptions.LatexifySyntaxError(
+                    "Multiple statements are not supported in Match nodes."
+                )
+            true_latex = self.visit(match_case.body[0])
+            cond_latex = self.visit(match_case.pattern)
+
+            if i < len(node.cases) - 1:  # no wildcard cases
+                if match_case.guard:
+                    cond_latex = self.visit(match_case.guard)
+                    subject_latex = ""  # getting variable from cond_latex
+                if not cond_latex:
+                    raise exceptions.LatexifySyntaxError(
+                        "Match subtrees must contain only one wildcard at the end."
+                    )
+                latex += true_latex + r", & \mathrm{if} \ " + cond_latex + r" \\ "
+            else:
+                if cond_latex:
+                    raise exceptions.LatexifySyntaxError(
+                        "Match subtrees must contain only one wildcard at the end."
+                    )
+                latex += true_latex + r", & \mathrm{otherwise}"
+        latex += r"\end{array} \right."
+        latex_final = latex.replace("subject_name", subject_latex)
+        return latex_final
+
+    def visit_MatchValue(self, node: ast.MatchValue) -> str:
+        """Visit a MatchValue node"""
+        latex = self.visit(node.value)
+
+        return "subject_name = " + latex
+
+    def visit_MatchAs(self, node: ast.MatchAs) -> str:
+        """Visit a MatchAs node"""
+        """If MatchAs is a wildcard, return 'otherwise' case, else throw error"""
```

Comment:
nit: should be combined with the line above to form a single, multiline docstring